tests/elemental3/generate_image.pm

```diff
@@ -58,16 +63,41 @@ sub run {
     file_content_replace($config_file, '%TEST_PASSWORD%' => $rootpwd);
     assert_script_run("chmod 755 $config_file");
 
+    # Define architecture for the system extensions
+    given ($arch) {
+        when ('aarch64') {
+            $sysext_arch = 'arm64';
+        }
+        when ('x86_64') {
+            $sysext_arch = 'x86-64';
+        }
+    }
+
+    # Get the system extensions list
+    my @list = split(/[\r\n]+/, script_output("curl -s $sysext_path | sed -n 's/.*>\\(.*-.*-.*${sysext_arch}.raw\\)<.*/\\1/p'"));
```

Comment:
I'm sure that the `sed` command could be replaced by pur Perl code, but it was already in one of my script and it works. Maybe I will replace it later, if I have enough (free!) time for this.